lib/active_hash/base.rb

```diff
@@ -205,7 +205,7 @@ def create!(attributes = {})
       end
 
       def all(options = {})
-        ActiveHash::Relation.new(self, @records || [], options[:conditions] || {})
+        ActiveHash::Relation.new(self, @records || [], options[:conditions].to_a || [])
```

Comment:
`nil.to_a == []` so the last `|| []` is no longer needed

Reply:
Good catch!